### ISSUE TITLE
Using shaders causing dolls not to appear

### ISSUE THREAD

**Issue by Gamerbolts** [GenerationsMod/Generations-Core#49]
Hello, I am using Fabric 1.20.1. I was messing around with the mod and placed a doll down and it was invisible. See image below
![image](https://github.com/user-attachments/assets/3eb33c1d-7bd5-4b66-8639-2aae14cadcfa)
This happens with ComplementaryReimagined and Unbound vr5.2.2 and SIldur's Vibrant v1.51 Extreme VL. However it does not happen with Chocapic13_V9. I am not sure if this is something you can fix or if I should set the shader options to something different.

**Comment by Extendoz** [comment]
Known issue, just don't use shaders until hopefully we get a fix. Or use shaders without the Gen resource packs

**Comment by Gamerbolts** [comment]
ok will do, thanks!

**Comment by Extendoz** [comment]
the resource Generation packs just got updated and idk if it fixed any issues with that

**Comment by Extendoz** [comment]
How did you get it to work with Chocapic13_V9?

**Comment by Gamerbolts** [comment]
I have no idea. I clicked through each shader I had and chocapic worked with nothing I did on my end

**Comment by Extendoz** [comment]
can you share the shader pack? lol

**Comment by Extendoz** [comment]
or was it Chocapic13_V9 zipped? and you have to unzip it to get the shaders out of it?

**Comment by Extendoz** [comment]
![2024-09-19_16 12 38](https://github.com/user-attachments/assets/3dc035fa-1749-4f34-a74d-20f8f89071bb)
this is what mine looks like with every Chocapic13_V9 shader

**Comment by Gamerbolts** [comment]
[Chocapic13_V9.zip](https://github.com/user-attachments/files/17065818/Chocapic13_V9.zip)
It's this one. Just the zip

**Comment by Extendoz** [comment]
ahh i see, that doesnt do anything because the actual shaders are inside that zip

**Comment by Gamerbolts** [comment]
lol well that makes me feel dumb... The others were fine right out of the box, didn't think to check that one.

**Comment by Extendoz** [comment]
lol i was testing it the same way and i was like wtf, nothing is changing at all lol

**Comment by Gamerbolts** [comment]
Went back to this and went to my resoucepacks in the modpack and did not see a Generations one. I did report this to the devs for the shaders and they said to report to mod author as they are probably using their own shaders and causing issues.

**Comment by Gamerbolts** [comment]
Now using 1-9 of the Generations data packs. All pokemon are invisible. The mod itself most blocks and dolls are invisible while using shaders. I really wish there was something that can be done about this. I talked with the dev for Shaders and they said it was on the mod, they were using integrated shaders that was causing issues.

**Comment by Dysth** [comment]
This has been reported several times in our Discord, so I'll clear it up here too.
This will be fixed in Minecraft version 1.21.x alongside our future update to Cobblemon 1.6 (when it releases), after our initial release. We are first releasing on 1.5.2 Cobblemon on 1.20.1.

It's a complicated issue, but to simplify it- it's mostly down to our texture layering (multiple textures per Pokemon) stacking on top of another and texture/geometry to be in a format Iris accepts which is made a bit easier on 1.21. We've attempted a hacky fix at this, but after about a week of getting it "almost" working, we decided to put it on pause as further work would take more time, and potentially being impossible in the current versions and just sucking up way too much time for nothing, as this is just a hacky workaround which would need completely redone for 1.21.x so we opted to put our time to polishing the rest of the mod to get it finished first.

In short, shader support with Iris will be a #1 priority after our initial release and we will have it soon; just nothing we can do about it right now

**Comment by Gamerbolts** [comment]
Thanks for clearing that up, apologies I am not on the discord. Not sure I'll be able to update my pack to 1.21 unfortunately a lot of core mods I have are still in 1.20.1

**Comment by Dysth** [comment]
No worries, plenty of time- Likely a couple weeks before we're ready for release, and I imagine general fixes + a 1.21 port will take ~month so a good time for creators to start planning things out

**Comment by Waterpicker** [comment]
Closing in favor of consolidating all iris shaderpacks related stuff in one issue. Link here. https://github.com/GenerationsMod/Generations-Core/issues/98
